src/Microsoft.AspNet.Mvc.Core/ApplicationModels/ActionModel.cs

```diff
@@ -74,6 +78,6 @@ public ActionModel([NotNull] ActionModel other)
 
         public List<ParameterModel> Parameters { get; private set; }
 
-        public AttributeRouteModel AttributeRouteModel { get; set; }
+        public List<IRouteConstraintProvider> RouteConstraints { get; private set; }
```

Comment:
These are usually exposed as interfaces and not concrete types. I see another one here that's also `List` instead of `IList`.